tui/src/filter.rs

```diff
@@ -78,31 +78,29 @@ impl Filter {
                             has_children: false,
                         });
                     }
-
                     stack.extend(node.children().map(|child| child.id()));
                 }
             }
-            self.items.sort_by(|a, b| a.node.name.cmp(&b.node.name));
+            self.items
+                .sort_unstable_by(|a, b| a.node.name.cmp(&b.node.name));
         }
-
         self.update_completion_preview();
     }
 
     fn update_completion_preview(&mut self) {
-        if self.search_input.is_empty() {
-            self.completion_preview = None;
-            return;
+        self.completion_preview = if self.items.is_empty() || self.search_input.is_empty() {
+            None
+        } else {
+            let input = self.search_input.iter().collect::<String>().to_lowercase();
+            self.items.iter().find_map(|item| {
+                let item_name_lower = &item.node.name.to_lowercase();
+                if item_name_lower.starts_with(&input) {
+                    Some(item_name_lower[input.len()..].to_string())
+                } else {
+                    None
+                }
+            })
         }
```

Comment:
```suggestion
        });
```

Alongside my other suggested change. A semicolon should be suffixed for good practice since this is a statement, rather than just an expression which happens to return a unit type.